server/src/test/java/org/elasticsearch/search/aggregations/pipeline/MaxBucketIT.java

```diff
@@ -475,4 +486,56 @@ public void testNested() throws Exception {
         assertThat(maxBucketValue.value(), equalTo(maxTermsValue));
         assertThat(maxBucketValue.keys(), equalTo(maxTermsKeys.toArray(new String[maxTermsKeys.size()])));
     }
+
+    /**
+     * https://github.com/elastic/elasticsearch/issues/33514
+     *
+     * This bug manifests as the max_bucket agg ("peak") being added to the response twice, because
+     * the pipeline agg is run twice.  This makes invalid JSON and breaks conversion to maps.
+     * The bug was caused by an UnmappedTerms being the chosen as the first reduction target.  UnmappedTerms
+     * delegate reduction to the first non-unmapped agg, which would reduce and run pipeline aggs.  But then
+     * execution returns to the UnmappedTerms and _it_ runs pipelines as well, doubling up on the values.
+     *
+     * Applies to any pipeline agg, not just max.
+     */
+    public void testFieldGetsWrittenOutTwice() throws Exception {
```

Comment:
Isn't this testing that the field does NOT get written twice?

Reply:
Ah, yes, the name should be adjusted.  Named it while it was failing, but that doesn't make sense now that it's fixed :)